src/app/reports/reportsTable/types.ts

```diff
@@ -0,0 +1,54 @@
+import { SortState } from "@/components/Tables/Table";
+import { ServerSideSortMethod } from "@/components/Tables/sortMethods";
+import { DbReportRow } from "@/databaseUtils";
+
+export type ReportsSortMethod = ServerSideSortMethod<DbReportRow>;
+export type ReportsSortState = SortState<ReportsTableRow, ReportsSortMethod>;
+
+export type GetReportsDataAndCountErrorType =
+    | "abortedFetchingReportsTable"
+    | "abortedFetchingReportsTableCount"
+    | "failedToFetchReportsTable"
+    | "failedToFetchReportsTableCount";
+
+export type GetReportsReturnType =
+    | {
+          data: {
+              reportsData: ReportsTableRow[];
+              count: number;
+          };
+          error: null;
+      }
+    | {
+          data: null;
+          error: { type: GetReportsDataAndCountErrorType; logId: string };
+      };
+
+export type GetReportsCountReturnType =
+    | {
+          data: null;
+          error: { type: GetReportsDataAndCountErrorType; logId: string };
+      }
+    | {
+          data: number;
+          error: null;
```

Comment:
```suggestion
        data: number;
        error: null;
          
      }
    | {
        data: null;
        error: { type: GetReportsDataAndCountErrorType; logId: string };
```

Reply:
Am a little confused as to what the change is here - what is the space for?

Reply:
Oh space is an accident, the change is to swap the order so that the null data is at the bottom

Reply:
Hmm I thought the pattern in the codebase was to put data first (independently of nullness) - I think I was corrected this in my first ticket. That being said, with a ctrl + f, I think we've done a bit of both in the codebase with maybe slightly more like what I've suggested. Though happy to switch the order if you think so though :)

Reply:
Sorry I worded that badly - data stays at the top of each object but the object which has null data should go in the bottom I think, so 
```
{
data: not null
error: null
}
| {
 data: null
 error: not null
}
```

Reply:
Aah I see what you mean now -- will go and change it

Reply:
Done!